arithmetization/src/main/java/net/consensys/linea/zktracer/module/ec_data/EcDataOperation.java

```diff
@@ -0,0 +1,354 @@
+  private static final EWord SECP256_K1N =
+      EWord.ofHexString("0xfffffffffffffffffffffffffffffffffffffffffffffffffffffffefffffc2f");
+
+  private final Wcp wcp;
+  private final Ext ext;
+
+  private final int contextNumber;
+  private final int contextNumberDelta;
+  private final Bytes input;
+
+  private final int ecType;
+  @Getter private final int nRows;
+  private int hurdleSwitchOffRow;
+
+  private final List<Boolean> comparisons;
+  private final List<Boolean> equalities;
+  private final List<Bytes> squares;
+  private final List<Bytes> cubes;
+  private final List<Bytes> limbs;
+
+  // WCP interaction
+  private final List<Bytes> wcpArg1;
+  private final List<Bytes> wcpArg2;
+  private final List<OpCode> wcpInst;
+  private final List<Boolean> wcpRes;
+
+  // EXT interaction
+  private final List<Bytes> extArg1;
+  private final List<Bytes> extArg2;
+  private final List<Bytes> extArg3;
+  private final List<OpCode> extInst;
+  private final List<Bytes> extRes;
+
+  // pairing-specific
+  private int thisIsNotOnG2Index;
+  private final int nPairings;
+
+  private static int addressToEcType(final Address target) {
+    if (target.equals(Address.ECREC)) {
+      return EC_RECOVER;
+    } else if (target.equals(Address.ALTBN128_ADD)) {
+      return EC_ADD;
+    } else if (target.equals(Address.ALTBN128_MUL)) {
+      return EC_MUL;
+    } else if (target.equals(Address.ALTBN128_PAIRING)) {
+      return EC_PAIRING;
+    } else {
+      throw new IllegalArgumentException("invalid EC address");
+    }
+  }
+
+  private static int ecTypeToNRows(int ecType, final Bytes input) {
+    return switch (ecType) {
+      case EC_RECOVER, EC_ADD -> 8;
+      case EC_MUL -> 6;
+      case EC_PAIRING -> input.size() / 16;
+      default -> throw new IllegalArgumentException("invalid EC type");
+    };
+  }
+
+  private static boolean isOnG2(Bytes arg) {
+    return true; // TODO:
+  }
+
+  private static boolean isInfinity() {
+    return false; // TODO:
+  }
+
+  private EcDataOperation(
+      Wcp wcp, Ext ext, int contextNumber, int previousContextNumber, int ecType, Bytes input) {
+    Preconditions.checkArgument(EC_TYPES.contains(ecType), "invalid EC type");
+
+    final int nRows = ecTypeToNRows(ecType, input);
+    final int minInputLength =
+        switch (ecType) {
+          case EC_MUL -> 96;
+          default -> 128;
+        };
+    if (input.size() < minInputLength) {
+      this.input = leftPadTo(input, minInputLength);
+    } else {
+      this.input = input;
+    }
+    this.contextNumber = contextNumber;
+    this.contextNumberDelta = contextNumber - previousContextNumber;
+    this.ecType = ecType;
+    this.nRows = nRows;
+    this.thisIsNotOnG2Index = -1;
+    if (ecType == EC_PAIRING) {
+      this.nPairings = input.size() / 192;
+    } else {
+      this.nPairings = 0;
+    }
+
+    this.comparisons =
+        IntStream.range(0, nRows / 2).mapToObj(i -> false).collect(Collectors.toList());
```

Comment:
May the `IntStream.range(...)` logic can be extracted ot a small private method with two parameters?